dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -69,22 +77,55 @@
             .setVisibility(PropertyAccessor.FIELD, Visibility.ANY)
             .findAndRegisterModules();
 
-    private AppSecrets readJson(final char[] chars) throws DotDataException {
+    char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
         try {
-            return jsonMapper.readValue(new String(chars), AppSecrets.class);
+            final byte [] bytes = jsonMapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
         } catch (IOException e) {
             throw new DotDataException(e);
         }
     }
 
-    private String toJsonAsString(final AppSecrets object) throws DotDataException {
+    AppSecrets readJson(final char[] chars) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)